returnn/datasets/generating.py

```diff
@@ -756,6 +757,34 @@ def generate_seq(self, seq_idx):
                            for i in range(i1, i2)])
     return DatasetSeq(seq_idx=seq_idx, features=features, targets=targets)
 
+  def get_all_tags(self):
+    """
+    :rtype: list[str]
+    """
+    return ["seq-%i" % i for i in range(self._num_seqs)]
```

Comment:
`self._num_seqs` might be `inf`. It should handle that correctly (fall back to the default behavior).

Reply:
What is the default behaviour? 

Does it even make sense for a dummy dataset to be initialized with `num_seqs=inf`?  
In `GeneratingDataset` it is per default `inf` but here it has to be set by the user.

Reply:
> What is the default behaviour?

Whatever `super` of this function does. Just check the code.

> Does it even make sense for a dummy dataset to be initialized with `num_seqs=inf`?

Yes it does. Why not? It's usually only used for testing. In any case, this is explicitly supported.